test/tensorflow_test.py

```diff
@@ -60,5 +89,15 @@ def testTensorFlowVariables(self):
 
     ray.worker.cleanup()
 
+  def testTensorFlowVariableCollision(self):
```

Comment:
What is this testing? Don't we want to test that we can set/get the weights for the two networks independently without any interaction? Can we add something like the following? (This should mostly work although the `assertEqual` and `assertNotEqual` methods probably need to be adapted to work with numpy arrays.

```python
# Note we use only one worker to ensure that all of the remote functions run on the same worker.
ray.init(num_workers=1)

ray.env.net_vars1 = ray.EnvironmentVariable(net_vars_initializer, net_vars_reinitializer)
ray.env.net_vars2 = ray.EnvironmentVariable(net_vars_initializer, net_vars_reinitializer)

@ray.remote
def get_vars1():
  return ray.env.net_vars1.set_weights()

@ray.remote
def get_vars2():
  return ray.env.net_vars2.set_weights()

@ray.remote
def set_vars1(weights):
  ray.env.net_vars1.set_weights(weights)

@ray.remote
def set_vars2(weights):
  ray.env.net_vars2.set_weights(weights)

weights1 = get_vars1.remote()
weights2 = get_vars2.remote()
self.assertNotEqual(weights1, weights2)

# Swap the weights.
set_vars2.remote(weights1)
set_vars1.remote(weights2)

# Get the new weights.
new_weights1 = get_vars1.remote()
new_weights2 = get_vars2.remote()
self.assertNotEqual(weights1, weights2)

# Check that the weights were swapped.
self.assertEqual(weights1, new_weights2)
self.assertEqual(weights2, new_weights1)
```